internal/quaiapi/transaction_args.go

```diff
@@ -247,12 +247,12 @@ func (args *TransactionArgs) ToMessage(globalGasCap uint64, baseFee *big.Int) (t
 func (args *TransactionArgs) toTransaction() *types.Transaction {
 	var data types.TxData
 	switch {
-	case args.MaxFeePerGas != nil:
+	default:
```

Comment:
Agreed. Will remove.

Reply:
@jdowning100 does the new `InternalToExternal` TX type need any logic here? Looks like `toTransaction()` is called in a few places where I would expect handle that new type.

Reply:
From what I can tell toTransaction() is only actually used in one place: FillTransaction() in quaiapi/api.go and this function has no other references. If we think FillTransaction() is useful I can update it to support the other transaction type (note that this also does not support ExternalTransaction)

Reply:
Maybe we can leave it for now? This is elucidating a whole slew of TX APIs (anything that uses `TransactionArgs`) which currently assume everything is an `InternalTx` type. I don't think it will cause undesired behavior, it just means we don't have API support for any TX type other than `InternalTx`... I'll make an issue to revisit this later.

Reply:
Oh yeah, and I'll remove the switch too, @kiltsonfire

Reply:
Issue created here: https://github.com/dominant-strategies/go-quai/issues/524